engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/ReplayStateMachine.java

```diff
@@ -164,28 +171,41 @@ void replayNextEvent() {
       }
 
     } catch (final RuntimeException e) {
-      final var replayException =
-          new ProcessingException("Unable to replay record", currentEvent, metadata, e);
-      recoveryFuture.completeExceptionally(replayException);
+      recoveryFuture.completeExceptionally(new RuntimeException("Failed to replay records", e));
```

Comment:
Can you add more context to the error message? Such as lastSourceEventPostion, lastReplayedPosition etc.